htdocs/core/lib/usergroups.lib.php

```diff
@@ -1722,3 +1722,603 @@ function show_actions_done_user($conf, $langs, $db, $filterobj, $objcon = '', $n
 		print $out;
 	}
 }
+
+/**
+ *    	Show html area with actions in messaging format.
+ *      Note: Global parameter $param must be defined.
+ *
+ * 		@param	Conf		       $conf		   Object conf
+ * 		@param	Translate	       $langs		   Object langs
+ * 		@param	DoliDB		       $db			   Object db
+ * 		@param	mixed			   $filterobj	   Filter on object Adherent|Societe|Project|Product|CommandeFournisseur|Dolresource|Ticket|... to list events linked to an object
+ * 		@param	Contact		       $objcon		   Filter on object contact to filter events on a contact
+ *      @param  int			       $noprint        Return string but does not output it
+ *      @param  string		       $actioncode     Filter on actioncode
+ *      @param  string             $donetodo       Filter on event 'done' or 'todo' or ''=nofilter (all).
+ *      @param  array              $filters        Filter on other fields
+ *      @param  string             $sortfield      Sort field
+ *      @param  string             $sortorder      Sort order
+ *      @return	string|void				           Return html part or void if noprint is 1
+ */
+function show_actions_messaging_user($conf, $langs, $db, $filterobj, $objcon = '', $noprint = 0, $actioncode = '', $donetodo = 'done', $filters = array(), $sortfield = 'a.datep,a.id', $sortorder = 'DESC')
```

Comment:
Instead of adding a new function show_actions_messaging_user(), can you instead extends the existing show_actions_messaging(). 
When more than 90% of the code will be the same, it is better to add a variant of the function (for example by adding parameters if necessary) instead of duplicating the existing function.